cypress/integration/Trades.spec.js

```diff
@@ -1,120 +1,90 @@
 
+    let trade = new Trade(from, to, willPay, willReceive).sell();
 
-    cy.get(tid("proxy-creation-summary"), {timeout: waitForTradeToFinish})
-      .then((value) => {
-        expect(value.text().trim()).to.eq('You have successfully created a Proxy')
-      });
+    expect(trade).to.receive(`${willReceive}.00000`);
 
-    cy.get(tid("congratulation-message", (tid("sold-token"), tid("token-amount-value"))), {
-      timeout: waitForTradeToFinish,
-    }).contains(`${ETH_AMOUNT_TO_SELL} ${base}`);
+    const finalization = trade
+      .acceptTerms()
+      .execute();
 
-    cy.get(tid("congratulation-message", tid("bought-token", tid("token-amount-value"))), {
-      timeout: waitForTradeToFinish,
-    }).contains(`${DAI_AMOUNT_TO_RECEIVE} ${quote}`);
+    const summary = finalization
+      .willCreateProxy()
+      .willCommitATrade();
 
-    cy.get(tid("congratulation-message", tid("final-price", tid("token-amount-value"))), {
-      timeout: waitForTradeToFinish,
-    }).contains(expectedPrice);
+    summary.expectProxyBeingCreated();
+    summary.expectBought(willReceive, to);
+    summary.expectSold(willPay, from);
+    summary.expectPriceOf(price)
   });
 
   it("ETH for ERC20 with proxy", () => {
-    const base = "ETH";
-    const quote = "DAI";
-    const ETH_AMOUNT_TO_SELL = '1';
-    const DAI_AMOUNT_TO_RECEIVE = '275';
-    const expectedPrice = `275 ${base}/${quote}`;
-
-    cy.get(tid("wallets-continue")).contains("Continue").click();
+    let from = 'ETH',
+      to = 'DAI',
+      willPay = '1',
+      willReceive = '280',
+      price = '280 ETH/DAI';
 
-    cy.get(tid("set-trade-from-amount"))
-      .find('input').type(ETH_AMOUNT_TO_SELL);
+    let trade = new Trade(from, to, willPay, willReceive).sell();
 
-    cy.get(tid("terms-and-conditions")).click({position: "topRight", force: true});
-    cy.get(tid("initiate-trade")).click();
+    expect(trade).to.receive(`${willReceive}.00000`);
 
-    cy.get(tid("new-trade")).click({timeout: waitForTradeToFinish});
+    let finalization = trade
+      .acceptTerms()
+      .execute();
 
-    cy.get(tid("set-trade-from-amount"))
-      .find('input').type(ETH_AMOUNT_TO_SELL);
+    let summary = finalization
+      .willCreateProxy()
+      .willCommitATrade();
 
-    //  NOTE: we need to click this exact spot to avoid downloading PDF file instead (link is in the button as well)
-    cy.get(tid("terms-and-conditions")).click({position: "topRight", force: true});
-    cy.get(tid("initiate-trade")).click();
+    summary.expectProxyBeingCreated();
+    summary.expectBought(willReceive, to);
+    summary.expectSold(willPay, from);
+    summary.expectPriceOf(price);
 
-    cy.get(tid("trade-with-builtin-proxy-creation"))
-      .should('not.exist');
+    newTrade();
 
-    cy.get(tid("trade-token-from"))
-      .find(tid("token-amount-value"))
-      .then((value) => {
-        expect(value.text().trim()).to.eq(`${ETH_AMOUNT_TO_SELL} ${base}`);
-      });
+    willPay = '1';
```

Comment:
do we need to reuse vars here? Maybe intoduce new ones instead? Like `willPayWithProxy` or something

Reply:
and then everything is `const`